Drop unused dependencies (kf6-kwayland isn't packaged, anyway)

Private targets are no longer implied starting with Qt 6.10.0.

--- CMakeLists.txt.orig	2024-10-09 17:59:04 UTC
+++ CMakeLists.txt
@@ -34,19 +34,12 @@ find_package(Qt6 ${QT_MIN_VERSION} CONFIG REQUIRED
     Gui
     Test
 )
-
-find_package(KF6 ${KF6_MIN_VERSION} COMPONENTS
-  CoreAddons
-  Wayland
-)
-set_package_properties(KF6CoreAddons PROPERTIES
-  TYPE REQUIRED
-  PURPOSE "For loading Wayland backend plugins"
-)
-set_package_properties(KF6Wayland PROPERTIES
-  TYPE OPTIONAL
-  PURPOSE "Required for KWin/KWayland backend"
-)
+if(Qt6_VERSION VERSION_GREATER_EQUAL 6.10.0)
+  find_package(Qt6 6.10.0 CONFIG REQUIRED
+    COMPONENTS
+      GuiPrivate
+  )
+endif()
 
 find_package(Wrapland ${PROJECT_VERSION} CONFIG)
 set_package_properties(Wrapland PROPERTIES
